core/src/main/java/org/keycloak/sdjwt/vp/KeyBindingJwtVerificationOpts.java

```diff
@@ -73,61 +75,124 @@ public String getAud() {
         return aud;
     }
 
-    public static Builder builder() {
-        return new Builder();
+    public static KeyBindingJwtVerificationOpts.Builder builder() {
+        return new KeyBindingJwtVerificationOpts.Builder();
     }
 
-    public static class Builder extends TimeClaimVerificationOpts.Builder<Builder> {
+    public static KeyBindingJwtVerificationOpts.Builder builder(Integer clockSkew) {
+        return new KeyBindingJwtVerificationOpts.Builder(clockSkew);
+    }
 
+    public static class Builder extends IssuerSignedJwtVerificationOpts.Builder {
         private boolean keyBindingRequired = true;
         protected boolean validateIssuedAtClaim = true;
-        private int allowedMaxAge = DEFAULT_ALLOWED_MAX_AGE;
         private String nonce;
         private String aud;
 
+        public Builder() {
+            super();
+        }
+
+        public Builder(Integer clockSkew) {
+            super(clockSkew);
+        }
+
         public Builder withKeyBindingRequired(boolean keyBindingRequired) {
             this.keyBindingRequired = keyBindingRequired;
             return this;
         }
 
-        public Builder withAllowedMaxAge(int allowedMaxAge) {
-            this.allowedMaxAge = allowedMaxAge;
-            return this;
+        public KeyBindingJwtVerificationOpts.Builder withNonceCheck(String expectedNonce) {
+            this.nonce = expectedNonce;
+            return withClaimCheck(IDToken.NONCE, expectedNonce, true);
         }
 
-        public Builder withNonce(String nonce) {
-            this.nonce = nonce;
-            return this;
+        @Override
+        public KeyBindingJwtVerificationOpts.Builder withAudCheck(String expectedAud) {
+            this.aud = expectedAud;
+            return (Builder) super.withAudCheck(expectedAud);
         }
 
-        public Builder withAud(String aud) {
-            this.aud = aud;
-            return this;
+        @Override
+        public KeyBindingJwtVerificationOpts.Builder withIatCheck(Integer allowedMaxAge) {
+            return (Builder) super.withIatCheck(allowedMaxAge);
+        }
+
+        @Override
+        public KeyBindingJwtVerificationOpts.Builder withIatCheck(boolean isCheckOptional) {
+            return (Builder) super.withIatCheck(isCheckOptional);
+        }
+
+        @Override
+        public KeyBindingJwtVerificationOpts.Builder withIatCheck(Integer allowedMaxAge, boolean isCheckOptional) {
+            return (Builder) super.withIatCheck(allowedMaxAge, isCheckOptional);
+        }
+
+        @Override
+        public KeyBindingJwtVerificationOpts.Builder withNbfCheck() {
```

Comment:
I think that my other comment (for the class `IssuerSignedJWTVerificationOpts.builder`) applies here as well. If we use parameterized types for the "builder", we do not need to repeat all these methods.

Reply:
Perhaps something like this might work https://github.com/keycloak/keycloak/pull/44227#discussion_r2539358453 ?